pkg/flavor/kubernetes/multiwriter.go

```diff
@@ -0,0 +1,673 @@
+func (flavor *Flavor) CreateNFSService(svcName string) error {
+	log.Tracef(">>>>> CreateNFSService with service name %s", svcName)
+	defer log.Tracef("<<<<< CreateNFSService")
+
+	// validate nfs service settings
+	if svcName == "" {
+		return fmt.Errorf("empty service name provided for creating nfs multiwriter service")
+	}
+
+	// check if nfs service already exists
+	getAction := func() error {
+		_, err := flavor.kubeClient.CoreV1().Services(multiWriterNamespace).Get(svcName, meta_v1.GetOptions{})
+		return err
+	}
+	exists, err := flavor.resourceExists(getAction, "service", svcName)
+	if err == nil && exists {
+		log.Infof("nfs service %s exists in %s namespace", svcName, multiWriterNamespace)
+		return nil
+	}
+
+	// create the nfs service
+	service := flavor.makeNFSService(svcName)
+	if _, err := flavor.kubeClient.CoreV1().Services(multiWriterNamespace).Create(service); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create nfs service %s, err %+v", svcName, err)
+		}
+		log.Infof("nfs service %s already exists", svcName)
+	} else {
+		log.Infof("nfs service %s started", svcName)
+	}
+	return nil
+}
+
+func (flavor *Flavor) GetNFSService(svcName string) (*core_v1.Service, error) {
+	log.Tracef(">>>>> GetNFSService with service name %s", svcName)
+	defer log.Tracef("<<<<< GetNFSService")
+
+	service, err := flavor.kubeClient.CoreV1().Services(multiWriterNamespace).Get(svcName, meta_v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return service, nil
+}
+
+// CreateNFSDeployment creates a nfs deployment with given name
+func (flavor *Flavor) CreateNFSDeployment(deploymentName, pvcName string) error {
+	log.Tracef(">>>>> CreateNFSDeployment with name %s", deploymentName)
+	defer log.Tracef("<<<<< CreateNFSDeployment")
+
+	// validate nfs deployment settings
+	if deploymentName == "" {
+		return fmt.Errorf("empty name provided for creating NFS multiwriter deployment")
+	}
+
+	// check if nfs deployment already exists
+	getAction := func() error {
+		_, err := flavor.kubeClient.AppsV1().Deployments(multiWriterNamespace).Get(deploymentName, meta_v1.GetOptions{})
+		return err
+	}
+	exists, err := flavor.resourceExists(getAction, "deployment", deploymentName)
+	if err == nil && exists {
+		log.Infof("nfs deployment %s exists in %s namespace", deploymentName, multiWriterNamespace)
+		return nil
+	}
+
+	// create a nfs deployment
+	deployment := flavor.makeNFSDeployment(deploymentName, pvcName)
+	if _, err := flavor.kubeClient.AppsV1().Deployments(multiWriterNamespace).Create(deployment); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create nfs deployment %s, err %+v", deploymentName, err)
+		}
+		log.Infof("nfs deployment %s already exists", deploymentName)
+		return nil
+	}
+	// make sure its available
+	err = flavor.waitForDeployment(deploymentName)
+	if err != nil {
+		return err
+	}
+	// successfully created nfs deployment. make sure its running.
```

Comment:
How do we make sure it's running?

Reply:
i am waiting for AvailableReplicas to be 1, only RUNNING pods will be available i think based on description..